app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeClient.java

```diff
@@ -184,12 +184,16 @@ public void addUserToChat(ChatRoom chatRoom, ChatMember member, TUMCabeVerificat
     }
 
     public Observable<ChatMessage> sendMessage(int roomId, TUMCabeVerification verification) {
-        //If the id is zero then its an new entry otherwise try to update it
-        int id = ((ChatMessage) verification.getData()).getId();
-        if (id == 0) {
+        ChatMessage message = (ChatMessage) verification.getData();
+        if (message == null) {
+            throw new IllegalStateException("TUMCabeVerification data is not a ChatMessage");
+        }
+
+        if (message.isNewMessage()) {
             return service.sendMessage(roomId, verification);
         }
-        return service.updateMessage(roomId, id, verification);
+
+        return service.updateMessage(roomId, message.getId(), verification);
```

Comment:
I think this can also be reduced into one API. I can simply check the sendMessage API if it has a message ID.